libs/core/langchain_core/runnables/graph_mermaid.py

```diff
@@ -238,9 +239,18 @@ def add_subgraph(edges: list[Edge], prefix: str) -> None:
     return mermaid_graph
 
 
-def _escape_node_label(node_label: str) -> str:
-    """Escapes the node label for Mermaid syntax."""
-    return re.sub(r"[^a-zA-Z-_0-9]", "_", node_label)
+def _sanitize_node_label(node_label: str) -> str:
+    """Convert a string into a Mermaid-compatible node label.
+
+    Any character outside [a-zA-Z0-9_-] is replaced with a 3-digit hash
+    fragment so that the output is safe to use as a Mermaid identifier.
+    """
+
+    def replacer(match: re.Match[str]) -> str:
+        non_allowed_ch = match.group(0)
+        return hashlib.md5(non_allowed_ch.encode("utf-8")).hexdigest()[:3]  # noqa: S324
```

Comment:
https://docs.astral.sh/ruff/rules/hashlib-insecure-hash-function/ -- we don't want to do `noqa` for things that violate bandit rules without a **very** good reason -- this will get flagged by security scanners